Sherlock/test.cc

```diff
@@ -185,18 +220,22 @@ TEST(Sherlock, SubscribeAsynchronously) {
   using namespace sherlock_unittest;
 
   auto bar_stream = current::sherlock::Stream<Record>();
+  current::time::SetNow(std::chrono::microseconds(40));
   bar_stream.Publish(4);
+  current::time::SetNow(std::chrono::microseconds(50));
   bar_stream.Publish(5);
+  current::time::SetNow(std::chrono::microseconds(60));
   bar_stream.Publish(6);
   Data d;
-  std::unique_ptr<SherlockTestProcessor> p(new SherlockTestProcessor(d, false));
+  std::unique_ptr<SherlockTestProcessor> p(new SherlockTestProcessor(d, false, true));
   p->SetMax(4u);
   bar_stream.Subscribe(std::move(p)).Detach();  // `.Detach()` results in the subscriber running on its own.
   while (d.seen_ < 3u) {
     ;  // Spin lock.
   }
   EXPECT_EQ(3u, d.seen_);
-  EXPECT_EQ("4,5,6", d.results_);  // No `TERMINATE` for an asyncronous subscriber.
+  std::vector<std::string> expected_values{"[0:40,2:60] 4", "[1:50,2:60] 5", "[2:60,2:60] 6"};
```

Comment:
The function accepts a copy by value. The following works w/ my `g++` and `clang++`:

```
 $ git diff
diff --git a/Sherlock/test.cc b/Sherlock/test.cc
index 176738b..7aef243 100644
--- a/Sherlock/test.cc
+++ b/Sherlock/test.cc
@@ -181,7 +181,7 @@ TEST(Sherlock, SubscribeAndProcessThreeEntries) {
   }
   ASSERT_FALSE(d.subscriber_alive_);
 
-  std::vector<std::string> expected_values{"[0:10,2:30] 1", "[1:20,2:30] 2", "[2:30,2:30] 3"};
+  const std::vector<std::string> expected_values{"[0:10,2:30] 1", "[1:20,2:30] 2", "[2:30,2:30] 3"};
   // A careful condition, since the subscriber may process some or all entries before going out of scope.
   EXPECT_TRUE(
       CompareValuesMixedWithTerminate(d.results_, expected_values, SherlockTestProcessor::kTerminateStr))
@@ -209,7 +209,7 @@ TEST(Sherlock, SubscribeSynchronously) {
     ;  // Spin lock.
   }
 
-  std::vector<std::string> expected_values{"[0:40,2:60] 4", "[1:50,2:60] 5", "[2:60,2:60] 6"};
+  const std::vector<std::string> expected_values{"[0:40,2:60] 4", "[1:50,2:60] 5", "[2:60,2:60] 6"};
   // A careful condition, since the subscriber may process some or all entries before going out of scope.
   EXPECT_TRUE(
       CompareValuesMixedWithTerminate(d.results_, expected_values, SherlockTestProcessor::kTerminateStr))
@@ -234,7 +234,7 @@ TEST(Sherlock, SubscribeAsynchronously) {
     ;  // Spin lock.
   }
   EXPECT_EQ(3u, d.seen_);
-  std::vector<std::string> expected_values{"[0:40,2:60] 4", "[1:50,2:60] 5", "[2:60,2:60] 6"};
+  const std::vector<std::string> expected_values{"[0:40,2:60] 4", "[1:50,2:60] 5", "[2:60,2:60] 6"};
   EXPECT_EQ(Join(expected_values, ','), d.results_);  // No `TERMINATE` for an asyncronous subscriber.
   EXPECT_TRUE(d.subscriber_alive_);
   bar_stream.Publish(42);  // Need the 4th entry for the async subscriber to terminate.
@@ -477,7 +477,7 @@ TEST(Sherlock, ParsesFromFile) {
   }
   ASSERT_FALSE(d.subscriber_alive_);
 
-  std::vector<std::string> expected_values{"[0:100,2:300] 1", "[1:200,2:300] 2", "[2:300,2:300] 3"};
+  const std::vector<std::string> expected_values{"[0:100,2:300] 1", "[1:200,2:300] 2", "[2:300,2:300] 3"};
   // A careful condition, since the subscriber may process some or all entries before going out of scope.
   EXPECT_TRUE(
       CompareValuesMixedWithTerminate(d.results_, expected_values, SherlockTestProcessor::kTerminateStr))
```